lib/signaling/participant.js

```diff
@@ -198,14 +208,30 @@ class ParticipantSignaling extends StateMachine {
    * @returns {boolean}
    */
   connect(sid, identity) {
-    if (this.state === 'connecting') {
-      this._sid = sid;
-      this._identity = identity;
+    if (this.state === 'connecting' || this.state === 'reconnecting') {
+      if (!this._sid) {
+        this._sid = sid;
+      }
```

Comment:
```suggestion
      } else {
         // we must not get different sid on reconnection.
         assert(this._sid === sid);
      }
```

Reply:
@makarandp0 I don't think this is necessary, since we're sure that the SID won't change.

Reply:
I know..but I think assertions can help make this assumptions very explicit, and catch issues if they are broken.